nimbus/src/evaluator.rs

```diff
@@ -59,9 +80,13 @@ pub fn filter_enrolled(id: &Uuid, experiments: &[Experiment]) -> Result<Vec<Enro
                 slug: exp.arguments.slug.clone(),
                 user_facing_name: exp.arguments.user_facing_name.clone(),
                 user_facing_description: exp.arguments.user_facing_description.clone(),
-                branch_slug: choose_branch(&exp.arguments.slug, &exp.arguments.branches, id)?
-                    .clone()
-                    .slug,
+                branch_slug: choose_branch(
```

Comment:
IIUC branch bucketing should be solely based on the `nimbus_id`? Am I correct in this assumption?

Reply:
I had assumed that it would use the same id as the `filter_enrolled`, which would be either the nimbus_id, or whatever alternate "known" id was selected by the experiment. Maybe we should verify this is still the case with the current desktop/js implementation.

Reply:
Sounds like you're right: https://searchfox.org/mozilla-central/source/toolkit/components/messaging-system/experiments/ExperimentManager.jsm#201

Reply:
That's correct, assuming the `randomizationUnit` in the experiment's `bucketConfig` is `nimbus_id`, you should use value of `nimbus_id`.